lib/Sema/DerivedConformanceRawRepresentable.cpp

```diff
@@ -82,9 +82,16 @@ static void deriveBodyRawRepresentable_raw(AbstractFunctionDecl *toRawDecl) {
 
   Type rawTy = enumDecl->getRawType();
   assert(rawTy);
+  // If the raw type is a generic parameter, unwrap it to match the type of the literals.
+  auto sub_rawTy = dyn_cast<SubstitutedType>(rawTy.getPointer());
+  if (sub_rawTy && sub_rawTy->getReplacementType()->isTypeParameter()) {
+	  rawTy = sub_rawTy->getOriginal();
```

Comment:
It looks like you're trying to convert from an interface type to an ArchetypeType -- you should instead use ArchetypeBuilder::mapTypeIntoContext() for that